core/src/main/java/org/elasticsearch/search/builder/NewSearchSourceBuilder.java

```diff
@@ -0,0 +1,1077 @@
+        }
+        XContentBuilder builder = XContentFactory.jsonBuilder();
+        rescoreBuilder.toXContent(builder, EMPTY_PARAMS);
+        rescoreBuilders.add(builder.bytes());
+        return this;
+    }
+
+    public NewSearchSourceBuilder clearRescorers() {
+        rescoreBuilders = null;
+        return this;
+    }
+
+    /**
+     * Indicates whether the response should contain the stored _source for
+     * every hit
+     */
+    public NewSearchSourceBuilder fetchSource(boolean fetch) {
+        if (this.fetchSourceContext == null) {
+            this.fetchSourceContext = new FetchSourceContext(fetch);
+        } else {
+            this.fetchSourceContext.fetchSource(fetch);
+        }
+        return this;
+    }
+
+    /**
+     * Indicate that _source should be returned with every hit, with an
+     * "include" and/or "exclude" set which can include simple wildcard
+     * elements.
+     *
+     * @param include
+     *            An optional include (optionally wildcarded) pattern to filter
+     *            the returned _source
+     * @param exclude
+     *            An optional exclude (optionally wildcarded) pattern to filter
+     *            the returned _source
+     */
+    public NewSearchSourceBuilder fetchSource(@Nullable String include, @Nullable String exclude) {
+        return fetchSource(include == null ? Strings.EMPTY_ARRAY : new String[] { include }, exclude == null ? Strings.EMPTY_ARRAY
+                : new String[] { exclude });
+    }
+
+    /**
+     * Indicate that _source should be returned with every hit, with an
+     * "include" and/or "exclude" set which can include simple wildcard
+     * elements.
+     *
+     * @param includes
+     *            An optional list of include (optionally wildcarded) pattern to
+     *            filter the returned _source
+     * @param excludes
+     *            An optional list of exclude (optionally wildcarded) pattern to
+     *            filter the returned _source
+     */
+    public NewSearchSourceBuilder fetchSource(@Nullable String[] includes, @Nullable String[] excludes) {
+        fetchSourceContext = new FetchSourceContext(includes, excludes);
+        return this;
+    }
+
+    /**
+     * Indicate how the _source should be fetched.
+     */
+    public NewSearchSourceBuilder fetchSource(@Nullable FetchSourceContext fetchSourceContext) {
+        this.fetchSourceContext = fetchSourceContext;
+        return this;
+    }
+
+    /**
+     * Sets no fields to be loaded, resulting in only id and type to be returned
+     * per field.
+     */
+    public NewSearchSourceBuilder noFields() {
+        this.fieldNames = Collections.emptyList();
+        return this;
+    }
+
+    /**
+     * Sets the fields to load and return as part of the search request. If none
+     * are specified, the source of the document will be returned.
+     */
+    public NewSearchSourceBuilder fields(List<String> fields) {
```

Comment:
Yeah, personally I would prefer to use setter which set rather than append so I would be in favour of using this one.